powerstore/resource_filesystem.go

```diff
@@ -0,0 +1,637 @@
+			},
+			"smb_notify_on_change_dir_depth": schema.Int32Attribute{
+				Computed:            true,
+				Optional:            true,
+				Description:         "Lowest directory level to which the enabled notifications apply, if any.",
+				MarkdownDescription: "Lowest directory level to which the enabled notifications apply, if any.",
+				Validators: []validator.Int32{
+					int32validator.AtLeast(1),
+					int32validator.AtMost(512),
+				},
+			},
+			"is_async_mtime_enabled": schema.BoolAttribute{
+				Computed:            true,
+				Optional:            true,
+				Description:         "Indicates whether asynchronous MTIME is enabled on the file system or protocol snaps that are mounted writeable.",
+				MarkdownDescription: "Indicates whether asynchronous MTIME is enabled on the file system or protocol snaps that are mounted writeable.",
+			},
+			"protection_policy_id": schema.StringAttribute{
+				Computed:            true,
+				Optional:            true,
+				Description:         "Unique identifier of the protection policy applied to the file system.",
+				MarkdownDescription: "Unique identifier of the protection policy applied to the file system.",
+			},
+			"file_events_publishing_mode": schema.StringAttribute{
+				Optional:            true,
+				Computed:            true,
+				Description:         "State of the event notification services for all file systems of the NAS server.",
+				MarkdownDescription: "State of the event notification services for all file systems of the NAS server.",
+				PlanModifiers: []planmodifier.String{
+					DefaultAttribute("None"),
+				},
+				Validators: []validator.String{
+					stringvalidator.OneOf([]string{
+						"None",
+						"SMB_Only",
+						"NFS_Only",
+						"All",
+					}...),
+				},
+			},
+			"host_io_size": schema.StringAttribute{
+				Computed:            true,
+				Optional:            true,
+				Description:         "Typical size of writes from the server or other computer using the VMware file system to the storage system.",
+				MarkdownDescription: "Typical size of writes from the server or other computer using the VMware file system to the storage system.",
+				Validators: []validator.String{
+					stringvalidator.OneOf([]string{
+						"VMware_8K",
+						"VMware_16K",
+						"VMware_32K",
+						"Vmware_64K",
+					}...),
+				},
+			},
+			"file_system_type": schema.StringAttribute{
+				Computed:            true,
+				Description:         "Type of filesystem: normal or snapshot.",
+				MarkdownDescription: "Type of filesystem: normal or snapshot.",
+			},
+			"flr_attributes": schema.SingleNestedAttribute{
+				Optional:            true,
+				Computed:            true,
+				Description:         "Type of filesystem: normal or snapshot.",
+				MarkdownDescription: "Type of filesystem: normal or snapshot.",
+				Attributes: map[string]schema.Attribute{
+					"mode": schema.StringAttribute{
+						Description:         "The FLR type of the file system.",
```

Comment:
mode has only 3 values, so can we add a validation for the same

Reply:
Done